src/storage/Storage.js

```diff
@@ -440,6 +440,94 @@ class Storage extends EventEmitter {
             }
         })
     }
+
+    async getFirstMessageTimestampInStream(streamId, partition){
+      const bucketQuery = 'SELECT id FROM bucket WHERE stream_id=? AND partition =? ORDER BY date_create ASC LIMIT 1'
+
+      const queryParams = [streamId, partition]
+
+      const buckets = await this.cassandraClient.execute(bucketQuery, queryParams, {
+          prepare: true,
+      })
+
+      if (buckets.rows.length !== 1) return 0
```

Comment:
Our style is currently to always have curly brackets around `if` and newline after (even with guard-clauses) so:
```js
if (condition) {
   return xxx
}
```